py/visdom/server.py

```diff
@@ -129,6 +130,40 @@ def hash_password(password):
     return hashlib.sha256(password.encode("utf-8")).hexdigest()
 
 
+
+
+class LazyEnvData(Mapping):
+    def __init__(self, env_path_file):
+        self._env_path_file = env_path_file
+        self._raw_dict = None
+
+    def _load_data(self):
+        try:
+            with open(self._env_path_file, 'r') as fn:
+                env_data = tornado.escape.json_decode(fn.read())
+        except Exception as e:
+            raise ValueError(
+                "Failed loading environment json: {} - {}".format(
+                    self._env_path_file, repr(e)))
+        self._raw_dict = {
+                'jsons': env_data['jsons'],
+                'reload': env_data['reload']
+        }
+
+    def __getitem__(self, key):
+        if not self._raw_dict:
+            self._load_data()
+        return self._raw_dict.__getitem__(key)
+
+    def __iter__(self):
+        self._load_data()
```

Comment:
Shouldn't this and the `__len__` implementation both mirror the `__getitem__` implementation in only loading if the load hasn't already happened? 

It may be cleaner to move the `None` check into `_load_data` and update it to only attempt the load if it hasn't already been done (after which, you could consider calling the function `_lazy_load_data` to better capture the fact that it doesn't always force a load).

Reply:
Good point, I've missed this!

I have changed the function `_load_data` as proposed and renamed the function to `lazy_load_data`.
(I feel that there is a utility in making it *public* in case some user needs to pre-fetch it in their script.)